src/Utils/signUpDetails.js

```diff
@@ -0,0 +1,14 @@
+export function signUpDetails(state, action) {
+  switch (action.type) {
+    case "FirstName":
+      return { ...state, firstName: action.firstName };
```

Comment:
Suggestion: Use action.payload as that is more conventional then action.firstName and so on.